front/src/applications/operationalStudies/views/Scenario/components/ManageTimetableItem/Itinerary/hooks/usePathStepsMetadata.ts

```diff
@@ -0,0 +1,234 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { skipToken } from '@reduxjs/toolkit/query';
+import type { Position } from 'geojson';
+
+import { useScenarioContext } from 'applications/operationalStudies/hooks/useScenarioContext';
+import useTrainPathOps from 'applications/operationalStudies/hooks/useTrainPathOps';
+import { osrdEditoastApi, type TrainSchedule } from 'common/api/osrdEditoastApi';
+import type { PathStepMetadata, PathStepV2 } from 'reducers/osrdconf/types';
+import { getPointOnTrackCoordinates } from 'utils/geometry';
+
+/**
+ * For each path step, get all its secondary codes and track names to display in the form
+ * and update the pathStepsMetadataById state.
+ */
+export const usePathStepsMetadata = (pathSteps: PathStepV2[]) => {
+  const { infraId } = useScenarioContext();
+
+  const { getTrackSectionsByIds } = useScenarioContext();
+
+  const [pathStepsMetadataById, setPathStepsMetadataById] = useState<Map<string, PathStepMetadata>>(
+    new Map()
+  );
+
+  // 1. Extract the train path to extract its steps related operational points
+  const strippedTrainPath: TrainSchedule['path'] = useMemo(
+    () =>
+      pathSteps.reduce<TrainSchedule['path']>((acc, step) => {
+        if (!step.location) return acc;
+        if ('uic' in step.location || 'trigram' in step.location) {
+          // strip location from its secondary_code so we can have all matchs for an uic or trigram
+          const {
+            secondary_code: _secCode,
+            track_reference: _trackRef,
+            ...location
+          } = step.location;
+          acc.push({
+            id: step.id,
+            ...location,
+          });
+          return acc;
+        }
+        acc.push({
+          id: step.id,
+          ...step.location,
+        });
+        return acc;
+      }, []),
+    [pathSteps]
+  );
+
+  const pathStepsOperationalPoints = useTrainPathOps(infraId, strippedTrainPath, {
+    returnAllOps: true,
+  });
+
+  // 2. Since a path step containing 'operation_point' as location will have only one match
+  // with postInfraByInfraIdMatchOperationalPoints, we need to call the endpoint again with
+  // the opId corresponding uic in order to get all the possible matches and have
+  // all the secondary codes and track names
+  const opIdsWithUic = useMemo(() => {
+    const opIds = pathSteps.reduce<string[]>((acc, step) => {
+      if (!step.location) return acc;
+      if ('operational_point' in step.location) {
+        acc.push(step.location.operational_point);
+      }
+      return acc;
+    }, []);
+
+    if (opIds.length === 0) return null;
```

Comment:
If all the uic are undefined, you will end up with an empty array and so the request should be skipped.
- maybe it would be better to return an empty array here
- you should change the condition to skip the request